src/app/CSE.KeyRotation/KeyRotationHelper.cs

```diff
@@ -52,8 +58,17 @@ private AsyncRetryPolicy GetCosmosRetryPolicy()
                         // Get the latest cosmos key.
                         Microsoft.Azure.KeyVault.Models.SecretBundle cosmosKeySecret = await keyVaultConnection.Client.GetSecretAsync(keyVaultConnection.Address, Constants.CosmosKey).ConfigureAwait(false);
 
+                        CosmosClientOptions options = new CosmosClientOptions
+                        {
+                            RequestTimeout = TimeSpan.FromSeconds(configuration.GetValue<int>(Constants.CosmosTimeout)),
+                            MaxRetryAttemptsOnRateLimitedRequests = configuration.GetValue<int>(Constants.CosmosRetry),
+                            MaxRetryWaitTimeOnRateLimitedRequests = TimeSpan.FromSeconds(configuration.GetValue<int>(Constants.CosmosTimeout)),
+                        };
+
+                        services.AddSingleton<CosmosClient>(new CosmosClient(configuration[Constants.CosmosUrl], cosmosKeySecret.Value));
```

Comment:
it seems strange to have this code here instead of startup or program.cs - is it possible to move it for consistency?

Reply:
This code was added here to replace the existing cosmos client object within the dependency container with the new one since there is a change in the key.